### ISSUE TITLE
the last class is nan

### ISSUE THREAD

**Issue by tjdgus6255** [open-mmlab/mmsegmentation#2146]
+-----------------------+-------+-------+
|         Class         |  IoU  |  Acc  |
+-----------------------+-------+-------+
|      _background_     | 16.65 |  83.2 |
|     circular_valve    |  0.0  |  0.0  |
|   fire_extinguisher   | 24.08 | 37.12 |
|     straight_valve    |  0.0  |  0.0  |
| fire_extinguisher_box | 15.48 | 21.92 |
|    emergency_light    |  0.09 |  0.09 |
|      flash_light      |  nan  |  nan  |
+-----------------------+-------+-------+

reudce_zero_label = True

num_classes = 7      -----> (I also tried 6, but I got a same error)

In data definition part, the last class is nan. 
--> When the last class was 'emergency_list', it also had nan.

**Comment by xiexinch** [comment]
Hi @tjdgus6255,
This may be related to the model you used and the distribution of your data. You could try a different network or assign weights to each class when calculating the loss.

**Comment by tjdgus6255** [comment]
Hello. 

According to your advice, I replaced the model (deeplabv3 -> pspnet) and dataset format(coco -> voc)

But I got a same issue.

I used the annotation tool of the link below. After annotation, we got annotation images (voc format)

annotation tool : https://github.com/wkentaro/labelme

(In fact, I still don't know what dataset format means... ex. kitti, voc, ...etc. Exactly, I'm not sure that formatting the data has meaningful difference)

**Comment by xiexinch** [comment]
Hi @tjdgus6255,
Could you provide more information like pytorch version, mmseg version, mmcv version, etc? Or you could try SegFormer-b2, it's much more robust.
And I think this should not be due to the data format.

**Comment by tjdgus6255** [comment]
Hello.

My environment here.. 
-------------------------------------------------
os                                ubuntu 18.04
python                         3.7.13 
pytorch                        1.10.1+cu111
mmcv-full                     1.6.0
mmsegmentation         0.25.0
-------------------------------------------------

**Comment by tjdgus6255** [comment]
+-----------------------+-------+-------+
|         Class         |  IoU  |  Acc  |
+-----------------------+-------+-------+
|   fire_extinguisher   | 23.05 | 38.18 |
|     straight_valve    | 23.95 | 42.54 |
| fire_extinguisher_box |  58.6 |  81.8 |
|    emergency_light    |  0.0  |  0.0  |
|      flash_light      | 48.59 | 78.15 |
|     circular_valve    |  0.08 |  0.08 |
|      _background_     |  nan  |  nan  |
+-----------------------+-------+-------+

In the dataset definition section, the first class was '_background_'. 

But I put it in the last part and trained. 

I think the last class is 'nan' due to 'reduce_zero_label=True' 

When I put 'reduce_zero_label=False', the last class was '0.0'

**Comment by wllzq126** [comment]
I met the some problem, then I changed some paramers in register data module。
before I wrote it like this:
    def __init__(self, **kwargs):
        super(guangfu, self).__init__(
            img_suffix='.JPG',
            seg_map_suffix='.png',
            ignore_index=255,
            **kwargs)
after I changed it like this:
    def __init__(self, **kwargs):
        super(guangfu, self).__init__(
            **kwargs
        )
Then the problem hadnt happened again.
Hope it can help you, besh wishes

**Comment by tjdgus6255** [comment]
> I met the some problem, then I changed some paramers in register data module。 before I wrote it like this: def **init**(self, **kwargs): super(guangfu, self).**init**( img_suffix='.JPG', seg_map_suffix='.png', ignore_index=255, **kwargs) after I changed it like this: def **init**(self, **kwargs): super(guangfu, self).**init**( **kwargs ) Then the problem hadnt happened again. Hope it can help you, besh wishes

Thanks for replying. I solved it. 

I replaced dataset_type('CustomDataset' -> 'PascalVOCDataset') and modified 'mmseg/evaluation/class_names.py', 'mmseg/datasets/voc.py'
